src/context_free.ml

```diff
@@ -478,6 +478,15 @@ class map_top_down ?(expect_mismatch_handler = Expect_mismatch_handler.nop)
       map_node EC.pattern pattern super#pattern x.ppat_loc base_ctxt x
 
     method! expression base_ctxt e =
+      let base_ctxt, e =
+        (* Make sure code-path attribute is applied before expanding. *)
+        match Attribute.get Ast_traverse.enter_value e with
+        | None -> (base_ctxt, e)
+        | Some { loc; txt } ->
+            ( Expansion_context.Base.enter_value ~loc txt base_ctxt,
+              Attribute.remove_seen Expression [ T Ast_traverse.enter_value ] e
+            )
+      in
```

Comment:
What about the `do_not_enter_let_module` attribute? Is it not important for that one to be handled before expanding as well? (see my comment above: I'm not really sure why it's important to apply some of the attributes before expanding)

Reply:
The only time it really matters to apply the attributes before expanding is when we're actually going to expand an extension node. Only let_module expressions can have do_not_enter_let_module attributes; extension nodes can't. So there's nothing to worry about with that attribute on extension nodes about to be expanded.